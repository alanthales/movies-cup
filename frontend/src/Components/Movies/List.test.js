import React from "react";
import MoviesList from "./List";
import renderer from "react-test-renderer";

const movies = [
  { id: "tt3606756", titulo: "Os Incríveis 2", ano: 2018, nota: 8.5 },
  { id: "tt4881806", titulo: "Jurassic World: Reino Ameaçado", ano: 2018, nota: 6.7 },
  { id: "tt5164214", titulo: "Oito Mulheres e um Segredo", ano: 2018, nota: 6.3 },
  { id: "tt5463162", titulo: "Deadpool 2", ano: 2018, nota: 8.1 }
];

test("render Movies List from given props", () => {
  const component = renderer.create(
    <MoviesList movies={movies} onSelect={() => {}} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});