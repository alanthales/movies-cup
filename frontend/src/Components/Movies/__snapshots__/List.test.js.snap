// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Movies List from given props 1`] = `
<div
  className="MuiGrid-root movies-component-list MuiGrid-container MuiGrid-justify-xs-space-between"
>
  <div
    className="MuiGrid-root movies-component-list-item MuiGrid-item"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-110 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <input
              className="PrivateSwitchBase-input-113"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
              value="tt3606756"
            />
          </span>
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <h5>
          Os Incríveis 2
        </h5>
        <p>
          2018
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root movies-component-list-item MuiGrid-item"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-110 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <input
              className="PrivateSwitchBase-input-113"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
              value="tt4881806"
            />
          </span>
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <h5>
          Jurassic World: Reino Ameaçado
        </h5>
        <p>
          2018
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root movies-component-list-item MuiGrid-item"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-110 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <input
              className="PrivateSwitchBase-input-113"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
              value="tt5164214"
            />
          </span>
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <h5>
          Oito Mulheres e um Segredo
        </h5>
        <p>
          2018
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root movies-component-list-item MuiGrid-item"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-110 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <input
              className="PrivateSwitchBase-input-113"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
              value="tt5463162"
            />
          </span>
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <h5>
          Deadpool 2
        </h5>
        <p>
          2018
        </p>
      </div>
    </div>
  </div>
</div>
`;
